dashboard/src/modules/components/DatePickerComponent/index.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from "react";
+import {
+  InputGroup,
+  InputGroupText,
+  DatePicker,
+  isValidDate,
+  yyyyMMddFormat,
+  Button,
+} from "@patternfly/react-core";
+import moment from "moment";
+
+function DatePickerWidget({
+  dataArray,
+  setPublicData,
+  controllerName,
+  setDateRange,
+}) {
+  const [fromDate, setFromDate] = useState(
+    moment(new Date(1990, 10, 4)).format("YYYY/MM/DD")
+  );
+  const [toDate, setToDate] = useState(
+    moment(new Date(2040, 10, 4)).format("YYYY/MM/DD")
+  );
+  const toValidator = (date) =>
+    isValidDate(fromDate) && date >= fromDate
+      ? ""
+      : "To date must be less than from date";
+  let modifiedArray = [];
+  const onFromChange = (_str, date) => {
+    setFromDate(new Date(date));
+    moment(date).format("DD/MM/YYYY");
+    if (isValidDate(date)) {
+      date.setDate(date.getDate() + 1);
+      setToDate(yyyyMMddFormat(date));
+    } else {
+      setToDate("");
+    }
+  };
+
+  const filterByDate = () => {
+    modifiedArray = dataArray.filter((data) => {
+      let formattedData = moment(data.metadata["dataset.created"]).format(
+        "YYYY/MM/DD"
+      );
+      return (
+        Date.parse(formattedData) >= Date.parse(fromDate) &&
+        Date.parse(formattedData) <= Date.parse(toDate) &&
+        data.name.includes(controllerName)
+      );
+    });
+    setPublicData(modifiedArray);
+    setDateRange(fromDate, toDate);
+  };
+  return (
+    <InputGroup style={{ marginLeft: "10px" }}>
+      <InputGroupText>Filter By Date</InputGroupText>
+      <DatePicker
+        onChange={onFromChange}
+        aria-label="Start date"
+        placeholder="YYYY-MM-DD"
+      />
+      <InputGroupText>to</InputGroupText>
+      <DatePicker
+        value={toDate}
+        onChange={(date) => setToDate(date)}
+        isDisabled={!isValidDate(fromDate)}
+        rangeStart={fromDate}
+        validators={[toValidator]}
+        aria-label="End date"
+        placeholder="YYYY-MM-DD"
+      />
+      <Button variant="control" onClick={filterByDate}>
+        Update
+      </Button>
```

Comment:
The isDisabled attribute on to-date picker is only a one time thing, which comes in effect when the dashboard initially loads, since at that time the from date is not selected (and hence it is an invalid date).
Bt other than that, what else can go wrong with the dates?

Reply:
> The isDisabled attribute on to-date picker is only a one time thing, which comes in effect when the dashboard initially loads, since at that time the from date is not selected (and hence it is an invalid date).

If we set the `toDate` to be "now" (or some other reasonable value, like maybe the newest dataset date), then it wouldn't be invalid and wouldn't need to be "disabled".

> Bt other than that, what else can go wrong with the dates?

I don't know....  A date in the future might be bad; a date which is "too old" might be bad; a date which doesn't exist (2022-02-29, or 2022-02-30, or 2022-14-01, or 2022-12-32, or something which is badly formatted or otherwise illegal) would be bad; it all depends....

Anyway, I just wanted you to give it some thought.  (And, I think you've done that.)